pkg/asset/installconfig/azure/client.go

```diff
@@ -473,3 +475,34 @@ func (c *Client) CheckIfExistsStorageAccount(ctx context.Context, resourceGroup,
 	}
 	return err
 }
+
+// GetRegionAvailabilityZones checks if a given region has availabililty zones for the nat gateways to use.
+func (c *Client) GetRegionAvailabilityZones(ctx context.Context, region string) (int, error) {
```

Comment:
Is there a reason to add a new function instead of using [the existing `GetAvailabilityZones` function](https://github.com/openshift/installer/blob/main/pkg/asset/installconfig/azure/client.go#L43)?